File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/InputsAndResultsTestByte.js
print("getByte")
if(typeof getByte !== 'undefined')
	throw "getByte already defined"
getByte = putByte
getByteType = typeof getByte
print(getByteType)
print(getByte)
if(getByteType !== 'number')
	throw "getByte not number!"

print("getByteVector")
if(typeof getByteVector !== 'undefined')
	throw "getByteVector already defined"
getByteVector = putByteVector
getByteVectorType = typeof getByteVector[0]
print(getByteVectorType)
print(getByteVector)
if(getByteVectorType !== 'number')
	throw "getByteVector not number!"

print("getByteVectorAsList")
if(typeof getByteVectorAsList !== 'undefined')
	throw "getByteVectorAsList already defined"
getByteVectorAsList = putByteVectorAsList
getByteVectorAsListType = typeof getByteVectorAsList[0]
print(getByteVectorAsListType)
print(getByteVectorAsList)
if(getByteVectorAsListType !== 'number')
	throw "getByteVectorAsList not number!"

print("getByteMatrix")
if(typeof getByteMatrix !== 'undefined')
	throw "getByteMatrix already defined"
getByteMatrix = putByteMatrix
getByteMatrixType = typeof getByteMatrix[0][0]
print(getByteMatrixType)
print(getByteMatrix)
if(getByteMatrixType !== 'number')
	throw "getByteMatrix not number!"

print("getByteMatrixAsList")
if(typeof getByteMatrixAsList !== 'undefined')
	throw "getByteMatrixAsList already defined"
getByteMatrixAsList = putByteMatrixAsList
getByteMatrixAsListType = typeof getByteMatrixAsList[0][0]
print(getByteMatrixAsListType)
print(getByteMatrixAsList)
if(getByteMatrixAsListType !== 'number')
	throw "getByteMatrixAsList not number!"

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/InputsAndResultsTestByte.rb
puts("getByte")
unless (defined?(getByte)).nil?
	raise "getByte already defined" + (defined?(getByte)).nil?
end
getByte = putByte
getByteType = getByte.class
puts(getByteType)
puts(getByte)
if(getByteType != 'number')
	raise "getByte not number!"
end

puts("getByteVector")
unless (defined?(getByteVector)).nil?
	raise "getByteVector already defined"
end
getByteVector = putByteVector
getByteVectorType = getByteVector[0].class
puts(getByteVectorType)
puts(getByteVector)
if(getByteVectorType != 'number')
	raise "getByteVector not number!"
end

puts("getByteVectorAsList")
unless (defined?(getByteVectorAsList)).nil?
	raise "getByteVectorAsList already defined"
end
getByteVectorAsList = putByteVectorAsList
getByteVectorAsListType = getByteVectorAsList[0].class
puts(getByteVectorAsListType)
puts(getByteVectorAsList)
if(getByteVectorAsListType != 'number')
	raise "getByteVectorAsList not number!"
end

puts("getByteMatrix")
unless (defined?(getByteMatrix)).nil?
	raise "getByteMatrix already defined"
end
getByteMatrix = putByteMatrix
getByteMatrixType = getByteMatrix[0][0].class
puts(getByteMatrixType)
puts(getByteMatrix)
if(getByteMatrixType != 'number')
	raise "getByteMatrix not number!"
end

puts("getByteMatrixAsList")
unless (defined?(getByteMatrixAsList)).nil?
	raise "getByteMatrixAsList already defined"
end
getByteMatrixAsList = putByteMatrixAsList
getByteMatrixAsListType = getByteMatrixAsList[0][0].class
puts(getByteMatrixAsListType)
puts(getByteMatrixAsList)
if(getByteMatrixAsListType != 'number')
	raise "getByteMatrixAsList not number!"
end
